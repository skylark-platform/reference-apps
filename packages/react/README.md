# `@skylark-reference-apps/react`

React Components for Skylark Reference Apps
