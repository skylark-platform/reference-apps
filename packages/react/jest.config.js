const base = require("../../jest.config");

module.exports = {
  ...base,
  setupFilesAfterEnv: ["./setupJestTests.js"],
};
