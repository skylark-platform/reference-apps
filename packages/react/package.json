{
  "name": "@skylark-reference-apps/react",
  "description": "Common React Components",
  "version": "1.0.0",
  "private": true,
  "main": "src/index.ts",
  "license": "UNLICENSED",
  "sideEffects": false,
  "scripts": {
    "test": "jest --config=jest.config.js",
    "lint": "eslint -c ../../.eslintrc.js ."
  },
  "devDependencies": {
    "@babel/core": "^7.16.0",
    "@babel/preset-env": "^7.16.4",
    "@babel/preset-react": "^7.16.0",
    "@babel/preset-typescript": "^7.16.0",
    "@storybook/react": "^6.3.12",
    "@testing-library/dom": "^8.11.1",
    "@testing-library/react": "^12.1.2",
    "@testing-library/user-event": "^13.5.0",
    "@types/jest": "^27.0.3",
    "eslint": "^8.3.0",
    "eslint-plugin-prettier": "^4.0.0",
    "jest": "^27.3.1",
    "jest-junit": "^13.0.0"
  },
  "dependencies": {
    "next": "12.1.4",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "react-icons": "^4.3.1"
  }
}
