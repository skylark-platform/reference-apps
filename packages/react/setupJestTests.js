// Extend Jest "expect" functionality with Testing Library assertions.
// eslint-disable-next-line import/no-extraneous-dependencies
import "@testing-library/jest-dom/jest-globals";

// eslint-disable-next-line global-require, @typescript-eslint/no-unsafe-return, import/no-extraneous-dependencies
jest.mock("next/router", () => require("next-router-mock"));
