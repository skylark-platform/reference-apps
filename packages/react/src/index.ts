export * from "./link";
export * from "./button";
export * from "./label";
export * from "./hello-world";
export * from "./typography";
export * from "./thumbnail";
