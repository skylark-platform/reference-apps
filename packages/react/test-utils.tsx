// eslint-disable-next-line eslint-comments/disable-enable-pair
/* eslint-disable import/no-extraneous-dependencies */
import React from "react";
import I18nProvider from "next-translate/I18nProvider";
import { render, RenderOptions } from "@testing-library/react";

import common from "@skylark-reference-apps/streamtv/locales/en-gb/common.json";

// eslint-disable-next-line react/prop-types
const AllTheProviders: React.JSXElementConstructor<{
  children: React.ReactElement;
}> = ({ children }) => (
  <I18nProvider lang="en-gb" namespaces={{ common }}>
    {children}
  </I18nProvider>
);

const customRender = (ui: React.ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
