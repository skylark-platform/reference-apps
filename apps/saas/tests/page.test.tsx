import { render } from "@testing-library/react";
import Home from "../pages/index";

describe("Calculator", () => {
  it("renders homepage unchanged", () => {
    const { container } = render(<Home />);
    expect(container).toMatchSnapshot();
  });
});
