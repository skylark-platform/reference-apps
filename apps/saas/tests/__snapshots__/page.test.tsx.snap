// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage renders homepage unchanged 1`] = `
<div>
  <div />
</div>
`;
